### PR TITLE
Render subchart notes on upgrade install

### PR DESCRIPTION
**Pull request by jgleonard**
I noticed that subchart notes weren't rendered when using `upgrade --install`.  This PR includes the fix and a test for it.

### CODE REVIEW

File: cmd/helm/testdata/testcharts/chart-with-subchart-notes/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
```
Comment (hickeyma) [inline cmd/helm/testdata/testcharts/chart-with-subchart-notes/Chart.yaml]: Can you set this to v2? New chart format for Helm 3.

File: cmd/helm/testdata/testcharts/chart-with-subchart-notes/charts/subchart-with-notes/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
```
Comment (hickeyma) [inline cmd/helm/testdata/testcharts/chart-with-subchart-notes/charts/subchart-with-notes/Chart.yaml]: Can you set this to v2? New chart format for Helm 3.

File: cmd/helm/testdata/testcharts/chart-with-subchart-notes/requirements.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+dependencies:
+  - name: subchart-with-notes
+    version: 0.0.1
```
Comment (hickeyma) [inline cmd/helm/testdata/testcharts/chart-with-subchart-notes/requirements.yaml]: Can add to parent `Chart.yaml` file instead and remove this file (new Helm 3 format).

**Review by hickeyma — CHANGES_REQUESTED** [review]
Thanks for pushing this fix @jgleonard. It is looking good but just fir a few comments around the test data. In addition to the inline comments, can you also remove the value files as hey are not necessary for testing?

File: cmd/helm/testdata/testcharts/chart-with-subchart-notes/Chart.yaml (hunk 1 above)
Comment (jgleonard) [reply cmd/helm/testdata/testcharts/chart-with-subchart-notes/Chart.yaml]: Good call.  Just pushed.

File: cmd/helm/testdata/testcharts/chart-with-subchart-notes/requirements.yaml (hunk 3 above)
Comment (jgleonard) [reply cmd/helm/testdata/testcharts/chart-with-subchart-notes/requirements.yaml]: Good call.  Just pushed.

File: cmd/helm/testdata/testcharts/chart-with-subchart-notes/charts/subchart-with-notes/Chart.yaml (hunk 2 above)
Comment (jgleonard) [reply cmd/helm/testdata/testcharts/chart-with-subchart-notes/charts/subchart-with-notes/Chart.yaml]: Good call.  Just pushed.

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @jgleonard.

**Pull request merged**